### ISSUE TITLE
Auto-generate the release notes upon release

### ISSUE THREAD

**Issue by thinkingserious** [sendgrid/sendgrid-nodejs#813]
#### Issue Summary

For every release, we hand craft the release notes on GitHub, now we would like to automate this process. Please see the existing [release notes](https://github.com/sendgrid/sendgrid-nodejs/releases) for an example. The contents of the release notes are generated from the CHANGELOG.md file. [Here](https://developer.github.com/v3/repos/releases/#create-a-release) is the documentation for updating the release notes via the GitHub API. 

#### Acceptance Criteria
 

* A script, which when run, creates a release note update based on the last deployed release

**Comment by devmattrick** [comment]
I'm willing to give this a shot.

**Comment by devmattrick** [comment]
Thoughts @thinkingserious?

**Comment by Berkmann18** [comment]
Alternatively, you could use `semantic-release` which resolves this _and_ #796 with a `.releaserc` config like this:
```json

{
  "plugins": [
    "@semantic-release/commit-analyzer",
    "@semantic-release/release-notes-generator",
    ["@semantic-release/changelog", {
      "changelogFile": "CHANGELOG.md"
    }],
    "@semantic-release/npm",
    "@semantic-release/github",
    ["@semantic-release/git", {
      "message": "Release ${nextRelease.version} :tada: [skip ci]\n\n${nextRelease.notes}"
    }],
  ]
}
```
And this in the `.travis.yml`:
```yml
jobs:
  include:
    - stage: release
       node_js: 'node'
       script: skip
       deploy:
         provider: script
         skip_cleanup: true
         script: run semantic-release
stages:
  - name: release
    if: branch = master
```
And the relevant additions in `package.json` of course.

**Comment by childish-sambino** [comment]
Closing as we're now using internal tooling for this.
